inline.h

```diff
@@ -2039,123 +2010,87 @@ Perl_is_utf8_non_invariant_string(const U8* const s, STRLEN len)
 /*
-=for apidoc is_strict_utf8_string
-
-Returns TRUE if the first C<len> bytes of string C<s> form a valid
-UTF-8-encoded string that is fully interchangeable by any application using
-Unicode rules; otherwise it returns FALSE.  If C<len> is 0, it will be
-calculated using C<strlen(s)> (which means if you use this option, that C<s>
-can't have embedded C<NUL> characters and has to have a terminating C<NUL>
-byte).  Note that all characters being ASCII constitute 'a valid UTF-8 string'.
+=for apidoc      is_utf8_string
+=for apidoc_item is_utf8_string_loc
+=for apidoc_item is_utf8_string_loclen
+=for apidoc_item is_strict_utf8_string
+=for apidoc_item is_strict_utf8_string_loc
+=for apidoc_item is_strict_utf8_string_loclen
+=for apidoc_item is_c9strict_utf8_string
+=for apidoc_item is_c9strict_utf8_string_loc
+=for apidoc_item is_c9strict_utf8_string_loclen
+=for apidoc_item is_utf8_string_flags
+=for apidoc_item is_utf8_string_loc_flags
+=for apidoc_item is_utf8_string_loclen_flags
+
+These each return TRUE if the first C<len> bytes of string C<s> form a valid
+UTF-8 string for varying degrees of strictness, FALSE otherwise.  If C<len> is
+0, it will be calculated using C<strlen(s)> (which means if you use this
+option, that C<s> can't have embedded C<NUL> characters and has to have a
+terminating C<NUL> byte).  Note that all characters being ASCII constitute 'a
+valid UTF-8 string'.
+
+Some of the functions also return information about the string.  Those that
+have the suffix C<_loc> in their names have an extra parameter, C<ep>.  If that
+is not NULL, the function stores into it the location of how far it got in
+parsing C<s>.  If the function is returning TRUE, this will be a pointer to the
+byte immediately after the end of C<s>.  If FALSE, it will be the location of
+the first byte that fails the criteria.
+
+The functions that instead have the suffix C<_loclen> have a second extra
+parameter, C<el>.  They act as the plain C<_loc> functions do with their C<ep>
+parameter, but if C<el> is not null, the functions store into it the number of
+UTF-8 encoded characters found at the point where parsing stopped.  If the
+function is returning TRUE, this will be the full count of the UTF-8 characters
+in C<s>; if FALSE, it will be the count before the first invalid one.
+
+C<is_utf8_string> (and C<is_utf8_string_loc> and C<is_utf8_string_loclen>)
+consider Perl's extended UTF-8 to be valid.  That means that
+code points above Unicode, surrogates, and non-character code points are
+all considered valid by this function.  Problems may arise in interchange with
+non-Perl applications, or (unlikely) between machines with different word
+sizes.
+
+C<is_strict_utf8_string> (and C<is_utf8_string_loc> and
+C<is_utf8_string_loclen>) consider only Unicode-range (0 to 0x10FFFF) code
+points to be valid, with the surrogates and non-character code points invalid.
+This level of strictness is what is safe to accept from outside sources that
+use Unicode rules.
```

Comment:
The mentions of `is_utf8_string_loc` and `is_utf8_string_loclen` should be the "strict" names.

Reply:
Thanks, fixed